cmd/plugin/tail_writer_test.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"io"
+	"testing"
+)
+
+func TestTailWriterStringWritethough(t *testing.T) {
+	tests := []struct {
+		size int
+		data []string
+		want string
+	}{
+		{
+			size: 1,
+			data: []string{"abc"},
+			want: "c",
+		},
+		{
+			size: 2,
+			data: []string{"abc"},
+			want: "bc",
+		},
+		{
+			size: 3,
+			data: []string{"abcde"},
+			want: "cde",
+		},
+		{
+			size: 2,
+			data: []string{"abc\n"},
+			want: "c\n",
+		},
+		{
+			size: 5,
+			data: []string{"hello", "world"},
+			want: "world",
+		},
+	}
+
+	for _, tt := range tests {
+		tw := NewTailWriter(tt.size)
+		for _, d := range tt.data {
+			tw.Write([]byte(d))
+		}
+		if got := tw.String(); got != tt.want {
```

Comment:
I guess there is a reason for avoiding the `stretchr/testify` dependency?

```suggestion
		assert.Equal(t, tt.want, tw.String())
```

Reply:
Go assertion libraries [are not allowed at Google](https://google.github.io/styleguide/go/decisions#assert), see the document for explanation. I generally do not use them on my projects because there is no reason, I see no value in them, this does the same thing and is readable for everyone.

Reply:
Replaced my own function with the `cmp` library.

Reply:
Just would remove boiler-plate in simple cases and the error messages are fine but I'm not the go expert and I'm ok in reducing dependencies.